src/main/java/com/yahoo/bullet/tracing/FilterRule.java

```diff
@@ -36,7 +36,8 @@ public FilterRule(String input, Map configuration) throws ParsingException {
 
     @Override
     public boolean consume(BulletRecord record) {
-        if (!specification.isAcceptingData() || !specification.filter(record)) {
+        // If rule is expired, not accepting data or does not match filters, don't consume...
+        if (isExpired() || !specification.isAcceptingData() || !specification.filter(record)) {
```

Comment:
It might be over-optimizing, but you could add a boolean member variable to AbstractRule, maybe called "expired", and check this first in isExpired() before doing the system call.  I think the 'System.currentTimeMillis()' system call is going to be an order of magnitude slower than checking a member variable.  So you could avoid the System call for all the times isExpired() is called after the rule has expired but before the next tick occurs...